### ISSUE TITLE
SEGV with CONSTSUB in cvref

### ISSUE THREAD

**Issue by atoomic** [rurban/perl-compiler#305]
with  >= 5.22 only

```perl
my $gen = sub { sub () { 8 } };
my $sub = &$gen;
print qq/ok\n/
```

gdb:
```
0x003599e9 in Perl_pp_anoncode () at pp.c:512
512	    if (CvCLONE(cv))
(gdb) p cv
$1 = (CV *) 0x5788d8
(gdb) p (cv)->sv_any
$3 = (XPVCV *) 0x0
```

**Comment by atoomic** [comment]
testc.sh 305

**Comment by rurban** [comment]
Fixed with commit c15616f3812695f2cdb406cd53494d3c911bfae8
Author: Reini Urban <rurban@cpanel.net>
Date:   Sat Nov 14 09:02:19 2015 +0100

    C: improve anonconst and empty GV CV handling

    This fixes 5.22 [GH #246]
    and improves Attribute::Handlers (GH #171, t/testc.sh 176)

    With 5.22 the CvGV->NAME is very often empty, esp. with const subs.
    Use the GV NAME then.
